File: ios/Classes/utils/LogUtil.m
//
//  LogUtil.m
//  flutter_tencentad
//
//  Created by 郭维佳 on 2021/12/1.
//

#import <Foundation/Foundation.h>
#import "LogUtil.h"

@interface LogUtil()
@property(nonatomic,assign) BOOL isDebug;
@end


@implementation LogUtil

+ (instancetype)sharedInstance{
    static LogUtil *myInstance = nil;
    if(myInstance == nil){
        myInstance = [[LogUtil alloc]init];
    }
    return myInstance;
}

- (void)debug:(BOOL)isDebug{
    _isDebug = isDebug;
}

- (void)print:(NSString *)message{
    if(_isDebug){
        GLog(@"%@", message);
    }
}

@end

File: ios/Classes/utils/LogUtil.swift
//
//  LogUtil.swift
//  flutter_tencentad
//
//  Created by gstory on 2021/9/4.
//

import Foundation

public class LogUtil : NSObject{
   
    static let logInstance = LogUtil()
    
    private var isDebug: Bool = false
    
    func isShow(debug:Bool){
        self.isDebug = debug
    }
    
    func printLog<T>(message: T,
                     file: String = #file,
                     method: String = #function,
                     line: Int = #line)
    {
        if(self.isDebug == true){
            print("\((file as NSString).lastPathComponent)[\(line)], \(method): \(message)")
        }
    }
}
